source/scripts/disruption/disruptor.py

```diff
@@ -0,0 +1,389 @@
+        disruption.publication_period.end = to_timestamp(self.end)
+
+        if not self.impacted_obj:
+            return feed_message.SerializeToString()
+
+        # Impacts
+        impact = disruption.impacts.add()
+        impact.id = "impact_" + self.id + "_1"
+        enums_impact = gtfs_realtime_pb2.Alert.DESCRIPTOR.enum_values_by_name
+        impact.created_at = to_timestamp(datetime.datetime.utcnow())
+        impact.updated_at = impact.created_at
+        impact.severity.effect = enums_impact[self.impact_type].number
+        impact.severity.id = "impact id for " + self.impact_type
+        impact.severity.wording = "impact wording for " + self.impact_type
+        if self.impact_type == "NO_SERVICE":
+            impact.severity.priority = 10
+            impact.severity.color = "#FF0000"
+        else:
+            impact.severity.priority = 1
+            impact.severity.color = "#FFFF00"
+
+        # Application periods
+        application_period = impact.application_periods.add()
+        application_period.start = to_timestamp(self.start)
+        application_period.end = to_timestamp(self.end)
+
+        # PT object
+        type_col = {
+            "network": chaos_pb2.PtObject.network,
+            "stop_area": chaos_pb2.PtObject.stop_area,
+            "line": chaos_pb2.PtObject.line,
+            "line_section": chaos_pb2.PtObject.line_section,
+            "route": chaos_pb2.PtObject.route,
+            "stop_point": chaos_pb2.PtObject.stop_point,
+        }
+
+        ptobject = impact.informed_entities.add()
+        ptobject.uri = self.impacted_obj.uri
+        ptobject.pt_object_type = type_col.get(self.impacted_obj.type, chaos_pb2.PtObject.unkown_type)
+        if ptobject.pt_object_type == chaos_pb2.PtObject.line_section:
+            line_section = ptobject.pt_line_section
+            line_section.line.uri = self.impacted_obj.uri
+            line_section.line.pt_object_type = chaos_pb2.PtObject.line
+            pb_start = line_section.start_point
+            pb_start.uri = self.impacted_obj.start
+            pb_start.pt_object_type = chaos_pb2.PtObject.stop_area
+            pb_end = line_section.end_point
+            pb_end.uri = self.impacted_obj.end
+            pb_end.pt_object_type = chaos_pb2.PtObject.stop_area
+
+        # Message with one channel and two channel types: web
+        message = impact.messages.add()
+        message.text = self.message
+        message.channel.name = "web"
+        message.channel.id = "web"
+        message.channel.max_size = 250
+        message.channel.content_type = "html"
+        message.channel.types.append(chaos_pb2.Channel.web)
+
+        return feed_message.SerializeToString()
+
+    def publish(self, producer):
+        producer.publish(self.to_pb())
+
+    def delete(self, producer):
+        self.is_deleted = True
+        producer.publish(self.to_pb())
+
+
+def parse_args(parser, logger):
+    parser = argparse.ArgumentParser()
```

Comment:
next time, if you do argument parsing, you may want to have a look at `docopt`

Reply:
Thanks for the **Tips**. Next time I will be ready :rocket: